src/plugins/enzymes/src/FindEnzymesDialog.cpp

```diff
@@ -611,7 +652,7 @@ FindEnzymesDialog::FindEnzymesDialog(ADVSequenceObjectContext* advSequenceContex
     initSettings();
 
     auto vl = new QVBoxLayout();
-    enzSel = new EnzymesSelectorWidget();
+    enzSel = new EnzymesSelectorWidget(advSequenceContext);
```

Comment:
`advSequenceContext` may be destroyed before the widget. Use a shared pointer and do not show any counts if the object is destroyed (or switch to another object in the view)

Reply:
If I use `QSharedPointer` here, `advSequenceContext` will be destroyed with `EnzymesSelectorWidget` - but it, of course, shouldn't be. We need to rewrite all `advSequenceContext` usage in UGENE - quite a large issue, don't you think?

Also, I can't get the scenario where `advSequenceContext` do not exist when you open a dialog, because the only possible way to open this dialog is from Sequence View.

Reply:
I do not ask to fix other places, only not to increase tech-debt. 
The new code written must be written correctly.

Sequence view may have multiple sequences attached: each will have a separate sequence context.
Some of the sequence may be dynamically removed. In this case the context of the removed sequence is destroyed: so you can't cache it unprotected outside.

Reply:
OK